BREAKING: Rename Call & Event

Szegoo opened the pull request:
Attempts to solve: #11736

polkadot companion: https://github.com/paritytech/polkadot/pull/5915
cumulus companion: https://github.com/paritytech/cumulus/pull/1563

Polkadot address: 126X27SbhrV19mBFawys3ovkyBS87SGfYwtwa8J2FjHrtbmA

Szegoo commented:
WIP

Szegoo commented on frame/support/procedural/src/construct_runtime/expand/event.rs:
```diff
@@ -79,7 +79,7 @@ pub fn expand_outer_event(
 		)]
 		#[allow(non_camel_case_types)]
-		pub enum Event {
+		pub enum RuntimeEvent {
```
Before I go on and start renaming Event into RuntimeEvent, it would be helpful if someone could check if I am doing the right thing. The main change is inside this file, the other changes are just the renaming of the Event. Should I just continue renaming Event into RuntimeEvent, or is there a smarter way to do this?

KiChjang replied:
So far so good, but you should also rename some `Event`s to `PalletEvent` depending on where they are used, e.g. I would look closer at the `event_variants` binding and check to see if they require renaming `Event`s to `PalletEvent`s.

KiChjang commented on bin/node-template/pallets/template/src/mock.rs:
```diff
@@ -36,7 +36,7 @@ impl system::Config for Test {
 	type Lookup = IdentityLookup<Self::AccountId>;
 	type Header = Header;
-	type Event = Event;
+	type Event = RuntimeEvent;
```
What should happen here is that the associated type should also be renamed to `RuntimeEvent`, so that it's clear which `Event` type to use while configuring the pallet.

kianenigma commented:
Looks in the right direction so far!

Szegoo commented:
I have renamed the associated type to `RuntimeEvent`, also I have renamed the Events inside pallets to `PalletEvent`. I need to figure out why this is failing some CI checks(it seems like an error is not getting thrown when it should), but other than that I am probably finished with renaming the events. It would be now helpful to check if what I did is correct.

Also, should I rename the `Call` enum in another PR, since there are quite a lot of changes here, so it might be easier for you to review if I make a separate PR for that? There would not be any dependency between these PRs. Is this a good idea?

CC: @KiChjang @kianenigma

paritytech-cicd-pr commented:
The CI pipeline was cancelled due to failure one of the required jobs. 
 The job name - test-linux-stable 
 The job logs - https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/1741073

paritytech-cicd-pr commented:
The CI pipeline was cancelled due to failure one of the required jobs. 
 The job name - test-linux-stable 
 The job logs - https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/1741074

paritytech-cicd-pr commented:
The CI pipeline was cancelled due to failure one of the required jobs. 
 The job name - test-linux-stable 
 The job logs - https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/1741072

paritytech-cicd-pr commented:
The CI pipeline was cancelled due to failure one of the required jobs. 
 The job name - test-linux-stable 
 The job logs - https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/1741113

paritytech-cicd-pr commented:
The CI pipeline was cancelled due to failure one of the required jobs. 
 The job name - test-linux-stable 
 The job logs - https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/1741149

paritytech-cicd-pr commented:
The CI pipeline was cancelled due to failure one of the required jobs. 
 The job name - test-linux-stable 
 The job logs - https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/1741185

paritytech-cicd-pr commented:
The CI pipeline was cancelled due to failure one of the required jobs. 
 The job name - test-linux-stable 
 The job logs - https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/1741221

paritytech-cicd-pr commented:
The CI pipeline was cancelled due to failure one of the required jobs. 
 The job name - test-linux-stable 
 The job logs - https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/1741265

paritytech-cicd-pr commented:
The CI pipeline was cancelled due to failure one of the required jobs. 
 The job name - test-linux-stable 
 The job logs - https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/1741301

paritytech-cicd-pr commented:
The CI pipeline was cancelled due to failure one of the required jobs. 
 The job name - test-linux-stable 
 The job logs - https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/1741372

Szegoo commented on frame/support/test/tests/construct_runtime_ui/undefined_event_part.stderr:
```diff
@@ -30,51 +29,19 @@ error[E0412]: cannot find type `Event` in module `pallet`
-58 - }
-   |
-
-error[E0412]: cannot find type `Event` in module `pallet`
-  --> tests/construct_runtime_ui/undefined_event_part.rs:49:1
-   |
-49 | / construct_runtime! {
-50 | |     pub enum Runtime where
-51 | |         Block = Block,
```
Not totally sure if this is the correct thing to do.

KiChjang commented on frame/support/procedural/src/construct_runtime/expand/event.rs:
```diff
@@ -30,7 +30,7 @@ pub fn expand_outer_event(
 	let mut query_event_part_macros = Vec::new();
 
 	for pallet_decl in pallet_decls {
-		if let Some(pallet_entry) = pallet_decl.find_part("Event") {
```
Sorry but I don't think this is necessary. The pallet part has always been called `Event`, and from context we know for sure that it's referring to the pallet event type, and not the runtime event type, so please revert this.

Szegoo replied:
Ok, so just to make sure I understand this correctly, the `Event` enum inside a pallet shouldn't be renamed to `PalletEvent`, but when accessing it from another pallet it should be accessed with `PalletEvent`?

KiChjang replied:
No, I think you misunderstood.

We _do not_ need to rename the `Event` pallet part. This here is where an `Event` pallet part is used:

```rust
construct_runtime!(
	pub enum Runtime where
		Block = Block,
		NodeBlock = Block,
		UncheckedExtrinsic = UncheckedExtrinsic,
	{
		System: frame_system::{Pallet, Call, Storage, Config, Event<T>},
	}
);
```

We _do_ need to rename the `Event` pallet enum to `PalletEvent`, i.e. whenever you see something like `#path::Event::<#runtime, #path::#inst>` or something similar to this type, you'll have to rename the `Event` there to `PalletEvent`.

In your latest push, you have completely removed all `PalletEvent`s. We need to add them back.

KiChjang commented on frame/support/procedural/src/construct_runtime/expand/event.rs:
```diff
@@ -39,19 +39,19 @@ pub fn expand_outer_event(
+				(Some(inst), true) => quote!(#path::PalletEvent::<#runtime, #path::#inst>),
+				(Some(inst), false) => quote!(#path::PalletEvent::<#path::#inst>),
+				(None, true) => quote!(#path::PalletEvent::<#runtime>),
+				(None, false) => quote!(#path::PalletEvent),
```
What is necessary however are these renames.

paritytech-cicd-pr commented:
The CI pipeline was cancelled due to failure one of the required jobs. 
 The job name - test-linux-stable 
 The job logs - https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/1742832

paritytech-cicd-pr commented:
The CI pipeline was cancelled due to failure one of the required jobs. 
 The job name - test-linux-stable 
 The job logs - https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/1742871

paritytech-cicd-pr commented:
The CI pipeline was cancelled due to failure one of the required jobs. 
 The job name - test-linux-stable 
 The job logs - https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/1748815

kianenigma reviewed (commented):
Looks really good! this is only refactoring the Event, right? 

We need `RuntimeCall` and `RutnimeOrigin` as well. Should we do those separately or here?

Szegoo commented:
> Looks really good! this is only refactoring the Event, right?
> 
> We need `RuntimeCall` and `RutnimeOrigin` as well. Should we do those separately or here?

Yes, this is only refactoring the Event for now. I will probably rename `RuntimeCall` and `RuntimeOrigin` in this PR.

Szegoo commented:
@kianenigma @KiChjang I believe that I am done with renaming the Events here. Maybe it would be better to make a separate PR for each rename, so could you please review this? I will look into resolving the errors in the companions.

kianenigma commented on bin/node-template/pallets/template/src/lib.rs:
```diff
@@ -42,7 +43,7 @@ pub mod pallet {
 	#[pallet::event]
 	#[pallet::generate_deposit(pub(super) fn deposit_event)]
-	pub enum Event<T: Config> {
+	pub enum PalletEvent<T: Config> {
```
@KiChjang some bikeshdding: I think this is rooting back to your initial suggestion [here](https://github.com/paritytech/substrate/issues/11736#issuecomment-1163169701) and having seen the outcome, I want to suggest keeping this enum called `Event` in the pallet and _ONLY_ prefixed with `Runtime` (i.e. `RuntimeEvet`) the outer enum. The reason being that within this code, we already know that this is inside a pallet, there's no need to emphasize again that this is the `Event` of a `Pallet`. For example, `pallet_balances::Event` makes a lot more sense to me than `pallet_balances::PalletEvent`. 

Moreover, if we call this `PalletEvent`, then one would ask why is `Event` called `PalletEvent` and `Origin` is not called `PalletOrigin`? 

Regardless of `Event` or `PalletEvent`, the current outcome is not consistent. All enums and structs inside a pallet should be prefixed with `Pallet`, not not prefixed at all and we should certainly chose one and stick to it (and clearly my preference is to ditch `Pallet` prefix). 

If you agree, it would be easy enough for @Szegoo to apply a blanket `PalletEvent` -> `Event` replacement script here.

kianenigma replied:
This would also lessen the diff size here.

KiChjang replied:
I had the same thought too, but I guess we'd still want some sort of differentiation between `Event` and `RuntimeEvent`... perhaps what we can do is keep it as `Event`, but create a type alias `type PalletEvent = Event` on the pallet itself during macro expansion?

Szegoo replied:
@KiChjang But if we make a type alias, when would we actually want to use the `Event` and when the `PalletEvent`, I assume that you meant to use the `Event` inside the pallet itself, and use `PalletEvent` outside the pallet. Wouldn't this be a bit confusing?

kianenigma replied:
I think the fewer keywords, the better, therefore not a fan of `PalletEvent` alias.

Szegoo replied:
Ok, so what is the final agreement then, should I change the `PalletEvent` back to `Event`?

KiChjang replied:
> I think the fewer keywords, the better, therefore not a fan of `PalletEvent` alias.

This is then saying that there's no need to differentiate between a runtime event and a pallet event? It is not a given that we always import the pallet event type by prepending `my_pallet::Event`. If we see a free-standing `Event` that's not part of a pallet module, do we then assume it's from a pallet event?

KiChjang replied:
> Ok, so what is the final agreement then, should I change the `PalletEvent` back to `Event`?

This should happen regardless. What we are discussing right now is another matter.

kianenigma replied:
> If we see a free-standing Event that's not part of a pallet module, do we then assume it's from a pallet event?

yeah, exactly.

Szegoo commented:
Since the diff size is now smaller I will rename Call and Origin here as well.

Szegoo commented:
@KiChjang @kianenigma I renamed the `Call` enum to `RuntimeCall`, could you please take a look?

ggwpez commented:
Please ask in the MR directly instead of opening SE questions in the future, thanks.

Szegoo commented:
> Please ask in the MR directly instead of opening SE questions in the future, thanks. For posterity: it was a flaky test.

Yeah, it is resolved now, I deleted the question

Szegoo commented:
@kianenigma @KiChjang Should I rename the `Origin` here as well?

Szegoo commented on frame/contracts/src/lib.rs:
```diff
@@ -243,13 +243,13 @@ pub mod pallet {
 
 		/// The overarching call type.
-		type Call: Dispatchable<Origin = Self::Origin, PostInfo = PostDispatchInfo>
+		type RuntimeCall: Dispatchable<Origin = Self::Origin, PostInfo = PostDispatchInfo>
```
Yes.

Szegoo commented on frame/examples/offchain-worker/src/lib.rs:
```diff
@@ -126,10 +126,10 @@ pub mod pallet {
 
 		/// The overarching dispatch call type.
-		type Call: From<Call<Self>>;
+		type RuntimeCall: From<Call<Self>>;
```
No, it is not :)

kianenigma commented:
@Szegoo your Polkadot address please, a tip is coming.

kianenigma reviewed (approved):
TYVM!

Szegoo commented:
> @Szegoo your Polkadot address please, a tip is coming.

I have added my address, thanks!

Szegoo commented:
@kianenigma Could you rerun the`gitlab-check-dependent-polkadot` CI check, it should be passing now.

Szegoo commented:
@shawntabrizi @bkchr Do you have an idea why is diener failing in `gitlab-check-dependent-polkadot`?

shawntabrizi commented:
@Szegoo gonna just bump all your prs, see if that fixes it

Szegoo commented:
@shawntabrizi Seems like merging with `master` didn't solve the problem :)

bkchr commented:
> @shawntabrizi @bkchr Do you have an idea why is diener failing in `gitlab-check-dependent-polkadot`?

You need to do `cargo update -p polkadot-runtime-common` in your Cumulus pr and push these changes.

shawntabrizi commented:
bot merge

shawntabrizi commented:
/tip medium

Szegoo commented:
@shawntabrizi Sorry for reminding you like this, but is the tip in #12145 still available?
